CRM/Volunteer/Form/Volunteer.php

```diff
@@ -90,16 +143,23 @@ function setDefaultValues() {
   * @return array of UFGroup (profile) IDs
   */
   private function getProfileIDs() {
-    if (empty($this->_profile_ids) && $this->_project !== FALSE) {
+    if (empty($this->_profile_ids) && $this->getProject() !== FALSE) {
       $dao = new CRM_Core_DAO_UFJoin();
       $dao->entity_table = CRM_Volunteer_BAO_Project::$_tableName;
-      $dao->entity_id = $this->_project->id;
+      $dao->entity_id = $this->getProject()->id;
       $dao->orderBy('weight asc');
       $dao->find();
       while ($dao->fetch()) {
         $this->_profile_ids[] = $dao->uf_group_id;
       }
     }
+    if (empty($this->_profile_ids)) {
```

Comment:
@GinkgoFJG pretty sure this default ensures backwards compat for existing volunteer projects... However, maybe we still need the upgrade step we discussed to cover upgrading the module and not updating the volunteer project? I suggest merging this and creating a new issue if so.